flutter_screen_recording` uses a deprecated version of the Android embedding

The plugin `flutter_screen_recording` uses a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if this plugin supports the Android V2 embedding. Otherwise, consider removing it since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.

Launching lib\main.dart on SM G990E in debug mode...

C:\flutter demo\video_demo\android>if "Windows_NT" == "Windows_NT" setlocal

C:\flutter demo\video_demo\android>set DEFAULT_JVM_OPTS=

C:\flutter demo\video_demo\android>set DIRNAME=C:\flutter demo\video_demo\android\

C:\flutter demo\video_demo\android>if "C:\flutter demo\video_demo\android\" == "" set DIRNAME=.

C:\flutter demo\video_demo\android>set APP_BASE_NAME=gradlew

C:\flutter demo\video_demo\android>set APP_HOME=C:\flutter demo\video_demo\android\

C:\flutter demo\video_demo\android>if defined JAVA_HOME goto findJavaFromJavaHome

C:\flutter demo\video_demo\android>set JAVA_HOME=C:\Program Files\Android\Android Studio\jbr

C:\flutter demo\video_demo\android>set JAVA_EXE=C:\Program Files\Android\Android Studio\jbr/bin/java.exe

C:\flutter demo\video_demo\android>if exist "C:\Program Files\Android\Android Studio\jbr/bin/java.exe" goto init

C:\flutter demo\video_demo\android>if not "Windows_NT" == "Windows_NT" goto win9xME_args

C:\flutter demo\video_demo\android>if "@eval[2+2]" == "4" goto 4NT_args

C:\flutter demo\video_demo\android>set CMD_LINE_ARGS=

C:\flutter demo\video_demo\android>set _SKIP=2

C:\flutter demo\video_demo\android>if "x-q" == "x" goto execute

C:\flutter demo\video_demo\android>set CMD_LINE_ARGS=-q -Ptarget-platform=android-arm64 "-Ptarget=C:\flutter demo\video_demo\lib\main.dart" -Pbase-application-name=android.app.Application -Pdart-defines=RkxVVFRFUl9XRUJfQVVUT19ERVRFQ1Q9dHJ1ZQ==,RkxVVFRFUl9XRUJfQ0FOVkFTS0lUX1VSTD1odHRwczovL3d3dy5nc3RhdGljLmNvbS9mbHV0dGVyLWNhbnZhc2tpdC9jZjdhOWQwODAwZjJhNWRhMTY2ZGJlMGViOWZiMjQ3NjAxODI2OWIxLw== -Pdart-obfuscation=false -Ptrack-widget-creation=true -Ptree-shake-icons=false -Pfilesystem-scheme=org-dartlang-root assembleDebug

C:\flutter demo\video_demo\android>goto execute

C:\flutter demo\video_demo\android>set CLASSPATH=C:\flutter demo\video_demo\android\\gradle\wrapper\gradle-wrapper.jar

C:\flutter demo\video_demo\android>"C:\Program Files\Android\Android Studio\jbr/bin/java.exe"    "-Dorg.gradle.appname=gradlew" -classpath "C:\flutter demo\video_demo\android\\gradle\wrapper\gradle-wrapper.jar" org.gradle.wrapper.GradleWrapperMain -q -Ptarget-platform=android-arm64 "-Ptarget=C:\flutter demo\video_demo\lib\main.dart" -Pbase-application-name=android.app.Application -Pdart-defines=RkxVVFRFUl9XRUJfQVVUT19ERVRFQ1Q9dHJ1ZQ==,RkxVVFRFUl9XRUJfQ0FOVkFTS0lUX1VSTD1odHRwczovL3d3dy5nc3RhdGljLmNvbS9mbHV0dGVyLWNhbnZhc2tpdC9jZjdhOWQwODAwZjJhNWRhMTY2ZGJlMGViOWZiMjQ3NjAxODI2OWIxLw== -Pdart-obfuscation=false -Ptrack-widget-creation=true -Ptree-shake-icons=false -Pfilesystem-scheme=org-dartlang-root assembleDebug
Parameter format not correct -

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class kotlin.collections.jdk8.CollectionsJDK8Kt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.internal.jdk8.JDK8PlatformImplementations found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.internal.jdk8.JDK8PlatformImplementations$ReflectSdkVersion found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.jvm.jdk8.JvmRepeatableKt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.jvm.optionals.OptionalsKt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.random.jdk8.PlatformThreadLocalRandom found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$1 found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$2 found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$3 found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.streams.jdk8.StreamsKt$asSequence$$inlined$Sequence$4 found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.text.jdk8.RegexExtensionsJDK8Kt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)
     Duplicate class kotlin.time.jdk8.DurationConversionsJDK8Kt found in modules jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0) and jetified-kotlin-stdlib-jdk8-1.7.10 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.7.10)

     Go to the documentation to learn how to <a href="d.android.com/r/tools/classpath-sync-errors">Fix dependency resolution errors</a>.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s


C:\flutter demo\video_demo\android>if "1" == "0" goto mainEnd

C:\flutter demo\video_demo\android>rem Set variable GRADLE_EXIT_CONSOLE if you need the _script_ return code instead of

C:\flutter demo\video_demo\android>rem the _cmd.exe /c_ return code!

C:\flutter demo\video_demo\android>if not "" == "" exit 1

C:\flutter demo\video_demo\android>exit /b 1
Exception: Gradle task assembleDebug failed with exit code 1


Exited (1).

--- a/.gitignore
+++ b/.gitignore
@@ -73,3 +73,4 @@
 
 .flutter-plugins
 .flutter-plugins-dependencies
+example/.flutter-plugins-dependencies

--- a/example/android/app/build.gradle
+++ b/example/android/app/build.gradle
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 32
+    compileSdkVersion 34
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'

--- a/example/android/app/src/main/AndroidManifest.xml
+++ b/example/android/app/src/main/AndroidManifest.xml
@@ -12,7 +12,7 @@
         </service>
 
         <activity
-            android:name=".MainActivity"
+            android:name="io.flutter.embedding.android.FlutterActivity"
             android:launchMode="singleTop"
             android:theme="@style/LaunchTheme"
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"

--- a/example/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
+++ b/example/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
@@ -5,7 +5,6 @@ import androidx.annotation.NonNull;
 import io.flutter.Log;
 
 import io.flutter.embedding.engine.FlutterEngine;
-import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
 
 /**
  * Generated file. Do not edit.
@@ -16,14 +15,13 @@ import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
 public final class GeneratedPluginRegistrant {
   private static final String TAG = "GeneratedPluginRegistrant";
   public static void registerWith(@NonNull FlutterEngine flutterEngine) {
-    ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
     try {
       flutterEngine.getPlugins().add(new com.pravera.flutter_foreground_task.FlutterForegroundTaskPlugin());
     } catch (Exception e) {
       Log.e(TAG, "Error registering plugin flutter_foreground_task, com.pravera.flutter_foreground_task.FlutterForegroundTaskPlugin", e);
     }
     try {
-      com.isvisoft.flutter_screen_recording.FlutterScreenRecordingPlugin.registerWith(shimPluginRegistry.registrarFor("com.isvisoft.flutter_screen_recording.FlutterScreenRecordingPlugin"));
+      flutterEngine.getPlugins().add(new com.isvisoft.flutter_screen_recording.FlutterScreenRecordingPlugin());
     } catch (Exception e) {
       Log.e(TAG, "Error registering plugin flutter_screen_recording, com.isvisoft.flutter_screen_recording.FlutterScreenRecordingPlugin", e);
     }

--- a/example/android/app/src/main/kotlin/com/isvisoft/flutter_screen_recording_example/MainActivity.kt
+++ b/example/android/app/src/main/kotlin/com/isvisoft/flutter_screen_recording_example/MainActivity.kt
@@ -1,6 +1,7 @@
 package com.isvisoft.flutter_screen_recording_example
 
 import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine;
 
 class MainActivity: FlutterActivity() {
 

--- a/flutter_screen_recording/android/build.gradle
+++ b/flutter_screen_recording/android/build.gradle
@@ -28,7 +28,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdkVersion 32
+    compileSdkVersion 34
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'

--- a/flutter_screen_recording/android/src/main/kotlin/com/isvisoft/flutter_screen_recording/FlutterScreenRecordingPlugin.kt
+++ b/flutter_screen_recording/android/src/main/kotlin/com/isvisoft/flutter_screen_recording/FlutterScreenRecordingPlugin.kt
@@ -3,35 +3,29 @@ package com.isvisoft.flutter_screen_recording
 import android.app.Activity
 import android.content.Context
 import android.content.Intent
-import android.content.res.Configuration
 import android.hardware.display.DisplayManager
 import android.hardware.display.VirtualDisplay
 import android.media.MediaRecorder
 import android.media.projection.MediaProjection
 import android.media.projection.MediaProjectionManager
 import android.os.Build
-import android.os.Environment
 import android.util.DisplayMetrics
 import android.util.Log
-import android.view.Display
-import android.view.WindowManager
 import androidx.core.app.ActivityCompat
-import androidx.core.content.ContextCompat.getSystemService
 import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
 import io.flutter.plugin.common.PluginRegistry
-import io.flutter.plugin.common.PluginRegistry.Registrar
-import java.io.File
 import java.io.IOException
 
 import com.foregroundservice.ForegroundService
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
 
 
-class FlutterScreenRecordingPlugin(
-        private val registrar: Registrar
-) : MethodCallHandler, PluginRegistry.ActivityResultListener{
+class FlutterScreenRecordingPlugin() : MethodCallHandler, PluginRegistry.ActivityResultListener, FlutterPlugin, ActivityAware {
 
     var mScreenDensity: Int = 0
     var mMediaRecorder: MediaRecorder? = null
@@ -48,23 +42,16 @@ class FlutterScreenRecordingPlugin(
 
     private lateinit var _result: MethodChannel.Result
 
+    var pluginBinding: FlutterPlugin.FlutterPluginBinding? = null
+    var activityBinding: ActivityPluginBinding? = null;
 
-    companion object {
-        @JvmStatic
-        fun registerWith(registrar: Registrar) {
-            val channel = MethodChannel(registrar.messenger(), "flutter_screen_recording")
-            val plugin = FlutterScreenRecordingPlugin(registrar)
-            channel.setMethodCallHandler(plugin)
-            registrar.addActivityResultListener(plugin)
-        }
-    }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
         if (requestCode == SCREEN_RECORD_REQUEST_CODE) {
             if (resultCode == Activity.RESULT_OK) {
                 mMediaProjectionCallback = MediaProjectionCallback()
                 mMediaProjection = mProjectionManager?.getMediaProjection(resultCode, data!!)
-                mMediaProjection?.registerCallback(mMediaProjectionCallback, null)
+                mMediaProjection?.registerCallback(mMediaProjectionCallback!!, null)
                 mVirtualDisplay = createVirtualDisplay()
                 _result.success(true)
                 return true
@@ -75,61 +62,69 @@ class FlutterScreenRecordingPlugin(
         return false
     }
 
-    override fun  onMethodCall(call: MethodCall, result: Result) {
-        if (call.method == "startRecordScreen") {
-            try {
-                _result = result
-                var title = call.argument<String?>("title")
-                var message = call.argument<String?>("message")
-                if (title == null || title == "") {
-                    title = "Your screen is being recorded";
-                }
-                if (message == null || message == "") {
-                    message = "Your screen is being recorded"
-                }
-                ForegroundService.startService(registrar.context(), title, message)
-                mProjectionManager = registrar.context().applicationContext.getSystemService(Context.MEDIA_PROJECTION_SERVICE) as MediaProjectionManager?
+    override fun onMethodCall(call: MethodCall, result: Result) {
+        val appContext = pluginBinding!!.applicationContext
 
-                val metrics = DisplayMetrics()
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
-                    mMediaRecorder = MediaRecorder(registrar.context().applicationContext)
-                } else {
-                    @Suppress("DEPRECATION")
-                    mMediaRecorder = MediaRecorder()
-                }
+        when (call.method) {
+            "startRecordScreen" -> {
+                try {
+                    _result = result
+                    var title = call.argument<String?>("title")
+                    var message = call.argument<String?>("message")
+                    if (title == null || title == "") {
+                        title = "Your screen is being recorded";
+                    }
+                    if (message == null || message == "") {
+                        message = "Your screen is being recorded"
+                    }
+                    ForegroundService.startService(appContext, title, message)
+                    mProjectionManager =
+                        appContext.getSystemService(
+                            Context.MEDIA_PROJECTION_SERVICE) as MediaProjectionManager?
 
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-                    val display = registrar.activity()!!.display
-                    display?.getRealMetrics(metrics)
-                } else {
-                    val defaultDisplay = registrar.context().applicationContext.getDisplay()
-                    defaultDisplay?.getMetrics(metrics)
-                }
-                mScreenDensity = metrics.densityDpi
-                calculeResolution(metrics)
-                videoName = call.argument<String?>("name")
-                recordAudio = call.argument<Boolean?>("audio")
-                startRecordScreen()
+                    val metrics = DisplayMetrics()
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                        mMediaRecorder = MediaRecorder(appContext)
+                    } else {
+                        @Suppress("DEPRECATION")
+                        mMediaRecorder = MediaRecorder()
+                    }
 
-            } catch (e: Exception) {
-                println("Error onMethodCall startRecordScreen")
-                println(e.message)
-                result.success(false)
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                        val display = activityBinding!!.activity.display
+                        display?.getRealMetrics(metrics)
+                    } else {
+                        val defaultDisplay = appContext.getDisplay()
+                        defaultDisplay?.getMetrics(metrics)
+                    }
+                    mScreenDensity = metrics.densityDpi
+                    calculeResolution(metrics)
+                    videoName = call.argument<String?>("name")
+                    recordAudio = call.argument<Boolean?>("audio")
+                    startRecordScreen()
+
+                } catch (e: Exception) {
+                    println("Error onMethodCall startRecordScreen")
+                    println(e.message)
+                    result.success(false)
+                }
             }
-        } else if (call.method == "stopRecordScreen") {
-            try {
-                ForegroundService.stopService(registrar.context())
-                if (mMediaRecorder != null) {
-                    stopRecordScreen()
-                    result.success(mFileName)
-                } else {
+            "stopRecordScreen" -> {
+                try {
+                    ForegroundService.stopService(appContext)
+                    if (mMediaRecorder != null) {
+                        stopRecordScreen()
+                        result.success(mFileName)
+                    } else {
+                        result.success("")
+                    }
+                } catch (e: Exception) {
                     result.success("")
                 }
-            } catch (e: Exception) {
-                result.success("")
             }
-        } else {
-            result.notImplemented()
+            else -> {
+                result.notImplemented()
+            }
         }
     }
 
@@ -145,7 +140,7 @@ class FlutterScreenRecordingPlugin(
         if (metrics.widthPixels > metrics.heightPixels) {
             var rate = metrics.widthPixels / maxRes
 
-            if(rate > 1.5){
+            if (rate > 1.5) {
                 rate = 1.5
             }
             mDisplayWidth = maxRes.toInt()
@@ -153,7 +148,7 @@ class FlutterScreenRecordingPlugin(
             println("Rate : $rate")
         } else {
             var rate = metrics.heightPixels / maxRes
-            if(rate > 1.5){
+            if (rate > 1.5) {
                 rate = 1.5
             }
             mDisplayHeight = maxRes.toInt()
@@ -172,7 +167,7 @@ class FlutterScreenRecordingPlugin(
     fun startRecordScreen() {
         try {
             try {
-                mFileName = registrar.context().getExternalCacheDir()?.getAbsolutePath()
+                mFileName = pluginBinding!!.applicationContext.externalCacheDir?.absolutePath
                 mFileName += "/$videoName.mp4"
             } catch (e: IOException) {
                 println("Error creating name")
@@ -195,12 +190,17 @@ class FlutterScreenRecordingPlugin(
             mMediaRecorder?.prepare()
             mMediaRecorder?.start()
         } catch (e: IOException) {
-            Log.d("--INIT-RECORDER", e.message+"")
+            Log.d("--INIT-RECORDER", e.message + "")
             println("Error startRecordScreen")
             println(e.message)
         }
         val permissionIntent = mProjectionManager?.createScreenCaptureIntent()
-        ActivityCompat.startActivityForResult(registrar.activity()!!, permissionIntent!!, SCREEN_RECORD_REQUEST_CODE, null)
+        ActivityCompat.startActivityForResult(
+            activityBinding!!.activity!!,
+            permissionIntent!!,
+            SCREEN_RECORD_REQUEST_CODE,
+            null
+        )
     }
 
     fun stopRecordScreen() {
@@ -211,7 +211,7 @@ class FlutterScreenRecordingPlugin(
             println("stopRecordScreen success")
 
         } catch (e: Exception) {
-            Log.d("--INIT-RECORDER", e.message +"")
+            Log.d("--INIT-RECORDER", e.message + "")
             println("stopRecordScreen error")
             println(e.message)
 
@@ -221,23 +221,23 @@ class FlutterScreenRecordingPlugin(
     }
 
     private fun createVirtualDisplay(): VirtualDisplay? {
-    try {
-      return mMediaProjection?.createVirtualDisplay(
-        "MainActivity", mDisplayWidth, mDisplayHeight, mScreenDensity,
-        DisplayManager.VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR, mMediaRecorder?.surface, null, null
-      )
-    } catch (e: Exception) {
-      println("createVirtualDisplay err")
-      println(e.message)
-      return null
-      }
+        try {
+            return mMediaProjection?.createVirtualDisplay(
+                "MainActivity", mDisplayWidth, mDisplayHeight, mScreenDensity,
+                DisplayManager.VIRTUAL_DISPLAY_FLAG_AUTO_MIRROR, mMediaRecorder?.surface, null, null
+            )
+        } catch (e: Exception) {
+            println("createVirtualDisplay err")
+            println(e.message)
+            return null
+        }
     }
 
     private fun stopScreenSharing() {
         if (mVirtualDisplay != null) {
             mVirtualDisplay?.release()
-            if (mMediaProjection != null) {
-                mMediaProjection?.unregisterCallback(mMediaProjectionCallback)
+            if (mMediaProjection != null && mMediaProjectionCallback != null) {
+                mMediaProjection?.unregisterCallback(mMediaProjectionCallback!!)
                 mMediaProjection?.stop()
                 mMediaProjection = null
             }
@@ -252,4 +252,25 @@ class FlutterScreenRecordingPlugin(
             stopScreenSharing()
         }
     }
+
+    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        pluginBinding = binding;
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
+
+    override fun onAttachedToActivity(binding: ActivityPluginBinding) {
+        activityBinding = binding;
+        val channel = MethodChannel(pluginBinding!!.binaryMessenger, "flutter_screen_recording")
+        channel.setMethodCallHandler(this)
+        activityBinding!!.addActivityResultListener(this);
+    }
+
+    override fun onDetachedFromActivityForConfigChanges() {}
+
+    override fun onReattachedToActivityForConfigChanges(binding: ActivityPluginBinding) {
+        activityBinding = binding;
+    }
+
+    override fun onDetachedFromActivity() {}
 }

